32blit-sdl/Main.cpp

```diff
@@ -233,13 +233,39 @@ int main(int argc, char *argv[]) {
 		SDL_SetWindowPosition(window, x, y);
 	}
 
+    blit_system = new System();
+    blit_input = new Input(blit_system);
+
+    printf("Found that many Joysticks: %d\n", SDL_NumJoysticks());
+
 	// Open all joysticks as game controllers
-	for(int n=0; n<SDL_NumJoysticks(); n++) {
-		SDL_GameControllerOpen(n);
+	for(int n = 0; n < SDL_NumJoysticks(); n++) {
+
+        if(SDL_IsGameController(n)) {
+
+            SDL_GameController* gc = SDL_GameControllerOpen(n);
+
+            if (gc != nullptr && SDL_GameControllerGetAttached(gc) == SDL_TRUE) {
```

Comment:
I'm not sure open can be successful if the controller isn't connected?

Reply:
Funnily, this is taken from some SDL Lib example. But I get your point, the second part of the if could probably be omited.